ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCaseBuilder.java

```diff
@@ -52,15 +56,54 @@ private TestCaseBuilder() {
     }
 
     try {
-      final Stream<Optional<String>> formats = test.formats().isEmpty()
-          ? Stream.of(Optional.empty())
-          : test.formats().stream().map(Optional::of);
-
-      final TestLocation location = testLocator.apply(test.name());
-
-      return formats
-          .map(format -> createTest(test, originalFileName, location, format))
-          .collect(Collectors.toList());
+      final List<TestCase> testCases = new LinkedList<>();
+
+      final List<Optional<String>> formats = test.formats().isEmpty()
+          ? Collections.singletonList(Optional.empty())
+          : test.formats().stream().map(Optional::of).collect(Collectors.toList());
+
+      final List<Optional<String>> configs;
+      final Entry<String, Object> overwrite;
+
+      if (test.config().isEmpty()) {
+        configs = Collections.singletonList(Optional.empty());
+        overwrite = null;
+      } else {
+        configs = test.config().stream().map(Optional::of).collect(Collectors.toList());
+
+        overwrite = Iterables.getOnlyElement(
+            test.properties().entrySet().stream()
```

Comment:
This assumes that the "config" field can only be used to configure properties, right? I don't love that this addition to the QTT framework feels very targeted / non-extensible, but if it's only meant as a short-term addition (with plans to be removed once it's served it's purpose), then this might be OK.

Reply:
> This assumes that the "config" field can only be used to configure properties, right?

Yes.

> I don't love that this addition to the QTT framework feels very targeted / non-extensible, but if it's only meant as a short-term addition (with plans to be removed once it's served it's purpose), then this might be OK.

Yeah, it's more or less a short-term hack. We can either remove it again, or make it more generic if we want. Given that we did discuss to maybe replace QTT, I did not want to go "all in" and make it super generic.